### ISSUE TITLE
[Feature Request] Ability to pause current rip

### ISSUE THREAD

**Issue by anonCoder123** [RipMeApp/ripme#9]
It would be nice to have the ability to pause the current url that is being ripped. We already have start and stop.

**Comment by metaprime** [comment]
I think adding pausing would be fine. I think that this would pause all ripping -- it would be a bit harder to pause one and have another start.

**Comment by rautamiekka** [comment]
^ I think he meant **pause all** anyway.

A way to have multiple rips running and pause any of them:
1) the GUI is expanded to show each thread/rip with its status, pause/resume and stop buttons. Without this, being certain of which thread you order to pause/resume or stop will be hard to predict. You can take example of what the GUI could look like from https://www.httrack.com when it's ripping (or mirroring, whichever term is appropriate (I have no idea)) a website.
2) each thread/rip is kept in a variable which has various data on it, like its status (what it's doing and whether it's running or paused, and if wanted/needed/..., whether it's been set to stop (the 'stop' status would be needed if the stopped threads/rips aren't deleted as they come (then they wouldn't need a separate 'stopped' status), but garbage-collected in intervals)).

**Comment by metaprime** [comment]
Yeah, I've thought about that kind of interface improvement. Especially when you consider that if you queue up a bunch of sites and many rippers delay a lot just because of rate limiting (which is specific to one site at a time), you realize that you could be running multiple rips (as long as they are different domains) at the same time. Ripme doesn't really have an interface for that.

**Comment by metaprime** [comment]
I think that would constitute a new issue.

**Comment by anonCoder123** [comment]
Yeah I didn't mean pause the current one and have another start. I meant like actual pause, and when you press "start" it would continue from where it left off.
